src/all/globalcomix/src/eu/kanade/tachiyomi/extension/all/globalcomix/GlobalComix.kt

```diff
@@ -0,0 +1,265 @@
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.plus
+import kotlinx.serialization.modules.polymorphic
+import okhttp3.CacheControl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+abstract class GlobalComix(final override val lang: String, private val extLang: String = lang) :
+    ConfigurableSource, HttpSource() {
+
+    override val name = "GlobalComix"
+    override val baseUrl = webUrl
+    override val supportsLatest = true
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private val json = Json {
+        isLenient = true
+        ignoreUnknownKeys = true
+        serializersModule += SerializersModule {
+            polymorphic(EntityDto::class) {
+                defaultDeserializer { UnknownEntity.serializer() }
+            }
+        }
+    }
+
+    private val intl = Intl(
+        language = lang,
+        baseLanguage = english,
+        availableLanguages = setOf(english),
+        classLoader = this::class.java.classLoader!!,
+        createMessageFileName = { lang -> Intl.createDefaultMessageFileName(lang) },
+    )
+
+    final override fun headersBuilder() = super.headersBuilder().apply {
+        set("Referer", "$baseUrl/")
+        set("Origin", baseUrl)
+        set("x-gc-client", clientId)
+        set("x-gc-identmode", "cookie")
+    }
+
+    override val client = network.client.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    private fun simpleQueryRequest(page: Int, orderBy: String?, query: String?): Request {
+        val url = apiSearchUrl.toHttpUrl().newBuilder()
+            .addQueryParameter("lang_id[]", extLang)
+            .addQueryParameter("p", page.toString())
+
+        orderBy?.let { url.addQueryParameter("sort", it) }
+        query?.let { url.addQueryParameter("q", it) }
+
+        return GET(url.build(), headers, CacheControl.FORCE_NETWORK)
+    }
+
+    override fun popularMangaRequest(page: Int): Request =
+        simpleQueryRequest(page, orderBy = null, query = null)
+
+    override fun popularMangaParse(response: Response): MangasPage =
+        mangaListParse(response)
+
+    override fun latestUpdatesRequest(page: Int): Request =
+        simpleQueryRequest(page, "recent", query = null)
+
+    override fun latestUpdatesParse(response: Response): MangasPage =
+        mangaListParse(response)
+
+    private fun mangaListParse(response: Response): MangasPage {
+        if (response.code == 204) {
+            return MangasPage(emptyList(), false)
+        }
+
+        val isSingleItemLookup = response.request.url.toString().startsWith(apiMangaUrl)
```

Comment:
the deep link request you make also uses apiMangaUrl, so how is this different?

Reply:
In searchMangaRequest, two different requests can be returned:
* If the query uses the pattern `id:slug-id-of-manga`, it uses `apiMangaUrl` to request the details for that single manga.
* Otherwise, it uses `apiSearchUrl` to do a normal search and requests a list of manga.

So I need to know which request was sent to choose the correct DTO.

Reply:
I didn't realize from a quick look at the time that both use different urls